src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -349,6 +382,35 @@ private static async Task<int> Run(string url, string hostname, string dockerHos
             {
                 builder.UseAzureRelay(options =>
                 {
+                    ClientCertificateCredential ccc = null;
+                    X509Store store = null;
+                    X509Certificate2 certificate = null;
+                    if (_certBasedAuth.HasValue())
+                    {
+                        for (int i = 1; i <= 8; i++)
+                        {
+                            store = new X509Store((StoreName)i, StoreLocation.LocalMachine);
+                            store.Open(OpenFlags.ReadOnly);
+                            foreach (var cert in store.Certificates)
```

Comment:
This for loop (and the others that do the same thing) can be replaced with a call to `store.Certificates.Find(X509FindType.FindByThumbprint, _certThumbprint.Value(), false)`

Reply:
Is there a reason that we would want to return invalid certificates?